geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveLayerReader.scala

```diff
@@ -0,0 +1,256 @@
+package geotrellis.spark.io.geowave
+
+import geotrellis.geotools._
+import geotrellis.proj4.LatLng
+import geotrellis.raster._
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.avro._
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.tiling.{LayoutDefinition, MapKeyTransform}
+import geotrellis.util._
+import geotrellis.vector.Extent
+
+import com.vividsolutions.jts.geom._
+import mil.nga.giat.geowave.adapter.raster.adapter.RasterDataAdapter
+import mil.nga.giat.geowave.core.geotime.store.query.IndexOnlySpatialQuery
+import mil.nga.giat.geowave.core.geotime.ingest._
+import mil.nga.giat.geowave.core.index.ByteArrayId
+import mil.nga.giat.geowave.core.store._
+import mil.nga.giat.geowave.core.store.index.CustomIdIndex
+import mil.nga.giat.geowave.core.store.operations.remote.options.DataStorePluginOptions
+import mil.nga.giat.geowave.core.store.query.QueryOptions
+import mil.nga.giat.geowave.datastore.accumulo._
+import mil.nga.giat.geowave.datastore.accumulo.metadata._
+import mil.nga.giat.geowave.mapreduce.input.{ GeoWaveInputKey, GeoWaveInputFormat }
+import org.apache.avro.Schema
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.geotools.coverage.grid._
+
+import spray.json._
+
+import scala.reflect._
+
+
+object GeowaveLayerReader {
+  val geometryFactory = new GeometryFactory
+  val p = geometryFactory.createPoint(new Coordinate(31.137720, 29.975307))
+  val emptySet = p.difference(p)
```

Comment:
We can most likely just account for the empty keybounds set a different way and not have these confusing magic numbers in here.